### ISSUE TITLE
what is the default early stopping with learner.autofit and how to fine-tune ReduceLROnPlataeu

### ISSUE THREAD

**Issue by semprepisfo** [amaiya/ktrain#464]
ReduceLROnPlataeu's patience is set to 2 according to what you wrote on the towards data science article but what is the default patience for EarlyStopping,

in the autofit I managed to insert the following values for ReduceRop
learner.autofit(2e-4, 140, callbacks = callbacks, reduce_on_plateau = 3, checkpoint_folder='/content', verbose=2, monitor='val_loss')

however the min_delta is not recognized

because of that so far I've been using :

callbacks = [EarlyStopping(monitor='val_loss', min_delta=1e-6, patience=1, verbose=2, mode='auto')]
learner.fit_onecycle(2e-5, 40, callbacks = callbacks) 

but I cannot insert the ReduceLROnPlataeu in the callback list, it throws an error: 

callbacks = [EarlyStopping(monitor='val_loss', min_delta=1e-6, patience=1, verbose=2, mode='auto'),
ReduceLROnPlateau(monitor="val_loss", factor=0.1, patience=1, verbose=2, min_delta=1e-6, min_lr = 1e-6, mode='auto')] 

learner.autofit(2e-5, 140, callbacks = callbacks)

**Comment by amaiya** [comment]
As stated in the docstring for `autofit`:
> If epochs is None, then early_stopping and reduce_on_plateau are atomatically set to 5 and 2, respectively.

If you want to fine-tune the callbacks to a greater degree, then you can just call `learner.fit` and supply whatever callbacks you want.  The cyclical learning rate in `autofit` (including its reduce LR behavior) is also implemented as a callback:  `ktrain.lroptimize.triangular.CycleLR`.  You can subclass/modify that class to fine-tune however you want and supply as callback to `learner.fit`.
